src/juju/watchers.ts

```diff
@@ -50,74 +73,30 @@ export function processDeltas(
     if (!modelWatcherData[modelUUID]) {
       modelWatcherData[modelUUID] = generateModelWatcherBase();
     }
-    switch (delta[0]) {
-      case "action":
-        switch (delta[1]) {
-          case "change":
-            const formatted = {
-              [delta[2].id]: delta[2],
-            };
-            mergeWith(modelWatcherData[modelUUID].actions, formatted);
-            break;
-        }
-        break;
-      case "application":
-        switch (delta[1]) {
-          case "change":
-            const formatted = {
-              [delta[2].name]: delta[2],
-            };
-            mergeWith(modelWatcherData[modelUUID].applications, formatted);
-            break;
-        }
-        break;
-      case "charm":
-        switch (delta[1]) {
-          case "change":
-            const formatted = {
-              [delta[2]["charm-url"]]: delta[2],
-            };
-            mergeWith(modelWatcherData[modelUUID].charms, formatted);
-            break;
-        }
-        break;
-      case "machine":
-        switch (delta[1]) {
-          case "change":
-            const formatted = {
-              [delta[2].id]: delta[2],
-            };
-            mergeWith(modelWatcherData[modelUUID].machines, formatted);
-            break;
-        }
-        break;
-      case "model":
-        switch (delta[1]) {
-          case "change":
-            mergeWith(modelWatcherData[modelUUID].model, delta[2]);
-            break;
-        }
-        break;
-      case "relation":
-        switch (delta[1]) {
-          case "change":
-            const formatted = {
-              [delta[2].key]: delta[2],
-            };
-            mergeWith(modelWatcherData[modelUUID].relations, formatted);
-            break;
-        }
-        break;
-      case "unit":
-        switch (delta[1]) {
-          case "change":
-            const formatted = {
-              [delta[2]["name"]]: delta[2],
-            };
-            mergeWith(modelWatcherData[modelUUID].units, formatted);
-            break;
-        }
-        break;
+    const modelData = modelWatcherData[modelUUID];
+    const _process = _processDelta.bind(null, delta[1], delta[2], modelData);
+    if (delta[0] === "action") {
+      _process("actions", delta[2].id);
+    }
+    if (delta[0] === "application") {
```

Comment:
`delta[0]` can't equal multiple entity types, so these could all be `else if`. Also here's another spot you could use the enums.